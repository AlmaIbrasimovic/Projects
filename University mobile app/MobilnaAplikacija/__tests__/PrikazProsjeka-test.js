
import  'react-native' ;
import React from 'react' ;
import SortGodine from '../November/SortiranjeGodina'
import renderer from 'react-test-renderer' ;

let findElement = function (tree, element) {
    let result = undefined;
    for(node in tree.children) {
        if(tree.children[node].props.testID==element) {
            result=true;
        }
    }
return result;
}


it('prosjeci po semestrima', () => {
    let tree  = renderer.create(
        <SortGodine />,
        {
            createNodeMock: (element) => {
              if (element.type === PersistGate) { return element.props.children }
              return null;
           }
        }
    ).toJSON();
        console.warn(tree);
    expect(findElement(tree, 'semestri')).toBeDefined();
})
it('prosjeci po godinama', () => {
    let tree  = renderer.create(
        <SortGodine />,
        {
            createNodeMock: (element) => {
              if (element.type === PersistGate) { return element.props.children }
              return null;
           }
        }
    ).toJSON();
        console.warn(tree);
    expect(findElement(tree, 'godine')).toBeDefined();
    //expect(tree).toMatchSnapshot();
})
