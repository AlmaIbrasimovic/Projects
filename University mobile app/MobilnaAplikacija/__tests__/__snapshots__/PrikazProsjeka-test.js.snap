// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prosjeci po semestrima 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
      "paddingTop": 15,
    }
  }
>
  <View
    testID="godine"
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
       Prosjeci po godinama
      

    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
       Prosjeci po godinama sortirani
      

       
    </Text>
  </View>
  <View
    testID="semestri"
  >
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
         Prosjeci po semestrima
        

      </Text>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "lightgrey",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginLeft": "5%",
          "marginRight": "5%",
          "marginTop": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <Text>
        Prika≈æi semestre sortirano po prosjeku
      </Text>
    </View>
  </View>
</View>
`;
